api/python/provisioner/commands/check.py

```diff
@@ -609,6 +609,106 @@ def _cluster_status(*, args: str) -> CheckEntry:
+
+        return res
+
+    @staticmethod
+    def _storage_luns(*, args: str) -> CheckEntry:
+        """Storage lvm check."""
+        res: CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
+        try:
+            from cortx.utils.kvstore.pillar import PillarStorage
+            from cortx.utils.validator.v_storage import StorageV
+        except ImportError:
+            res.set_fail(checked_target=cfg.ALL_MINIONS,
+                         comment="Package cortx-py-utils not installed")
+            return res
+        try:
+            nodes = PillarStorage().get("cluster:node_list")
+            StorageV().validate('luns', nodes)
+        except Exception as exc:
+            res.set_fail(checked_target=cfg.ALL_MINIONS,
+                         comment=str(exc))
+        else:
+            res.set_passed(checked_target=cfg.ALL_MINIONS)
+
+        return res
+
+    @staticmethod
+    def _mgmt_vip(*, args: str) -> CheckEntry:
+        """Network mgmt vip check."""
+        res: CheckEntry = CheckEntry(cfg.Checks.MGMT_VIP.value)
+        try:
+            from cortx.utils.kvstore.pillar import PillarStorage
+            from cortx.utils.validator.v_network import NetworkV
+        except ImportError:
+            res.set_fail(checked_target=local_minion_id(),
+                         comment="Package cortx-py-utils not installed")
+            return res
+        try:
+            mgmt_vip = PillarStorage().get("cluster:mgmt_vip")
+            NetworkV().validate('connectivity', [mgmt_vip])
+        except Exception as exc:
+            res.set_fail(checked_target=local_minion_id(),
+                         comment=str(exc))
+        else:
+            res.set_passed(checked_target=local_minion_id())
+
+        return res
+
+    @staticmethod
+    def _hostnames(*, args: str) -> CheckEntry:
+        """Validate hostnames check."""
+        res: CheckEntry = CheckEntry(cfg.Checks.HOSTNAMES.value)
+
+        try:
+            from cortx.utils.kvstore.pillar import PillarStorage
+            from cortx.utils.validator.v_network import NetworkV
+        except ImportError:
+            res.set_fail(checked_target=local_minion_id(),
+                         comment="Package cortx-py-utils not installed")
+            return res
+
+        res: List[CheckEntry] = list()
+        try:
+            check_salt: CheckEntry = CheckEntry(cfg.Checks.HOSTNAMES.value)
+            nodes = PillarStorage().get("cluster:node_list")
+
+            for node in nodes:
+                check_ent: CheckEntry = CheckEntry(cfg.Checks.HOSTNAMES.value)
+                try:
+                    hostname = PillarStorage().get(f"cluster:{node}:hostname")
+                    NetworkV().validate('connectivity', [hostname])
+                except Exception as exc:
+                    check_ent.set_fail(checked_target=local_minion_id(),
+                                       comment=str(exc))
+                check_ent.set_passed(checked_target=hostname)
```

Comment:
I think this line should be placed in `else`-part of **try/except** block. Otherwise, we will always trigger `set_passed`

Reply:
comment is addressed